internal/docstore/memdocstore/mem.go

```diff
@@ -91,35 +94,40 @@ func OpenCollection(keyField string, _ *Options) (*docstore.Collection, error) {
 // a document and returns the document's primary key. It should return nil if the
 // document is missing the information to construct a key. This will cause all
 // actions, even Create, to fail.
-func OpenCollectionWithKeyFunc(keyFunc func(docstore.Document) interface{}, _ *Options) (*docstore.Collection, error) {
-	c, err := newCollection("", keyFunc)
+func OpenCollectionWithKeyFunc(keyFunc func(docstore.Document) interface{}, opts *Options) (*docstore.Collection, error) {
+	c, err := newCollection("", keyFunc, opts)
 	if err != nil {
 		return nil, err
 	}
 	return docstore.NewCollection(c), nil
 }
 
-func newCollection(keyField string, keyFunc func(docstore.Document) interface{}) (driver.Collection, error) {
+func newCollection(keyField string, keyFunc func(docstore.Document) interface{}, opts *Options) (driver.Collection, error) {
 	if keyField == "" && keyFunc == nil {
 		return nil, gcerr.Newf(gcerr.InvalidArgument, nil, "must provide either keyField or keyFunc")
 	}
+	if opts == nil {
+		opts = &Options{RevisionField: docstore.RevisionField}
```

Comment:
I think you should create `&Options{}`, then check if `opts.RevisionField == ""` to set the default, in cases someone passes in a zero `Options`.

Reply:
Done.